### PR TITLE
Add tutorial doc for TF + TPU

### PR DESCRIPTION
**Pull request by Rocketknight1**
This is the sidebar tutorial for training with TF + TPU, to go with [the code notebook](https://github.com/huggingface/notebooks/pull/313).

Note that the Markdown is exported straight from Notion, so some formatting will probably look very wrong - I'm working on cleanup!

### CODE REVIEW

File: docs/source/en/perf_train_tpu_tf.mdx [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+When you use a **TPU Node**, you are effectively indirectly accessing a remote TPU. You will need a separate VM, which will initialize your network and data pipeline and then forward them to the remote node. When you use a TPU on Google Colab, you are accessing it in the **TPU Node** style.
+
+Using TPU Nodes can have some quite unexpected behaviour for people who aren’t used to them! In particular, because the TPU is located on a physically different system to the machine you’re running your Python code on, your data cannot be local to your machine - any data pipeline that loads from your machine’s internal storage will totally fail! Instead, data must be stored in Google Cloud Storage where your data pipeline can still access it, even when the pipeline is running on the remote TPU node.
+
+---
+
+**Tip:** If you can fit all your data in memory as `np.ndarray` or `tf.Tensor`, then you can `fit()` on that data even when using Colab or a TPU Node, without needing to upload it to Google Cloud Storage.
+
+---
```
Comment (sgugger) [inline docs/source/en/perf_train_tpu_tf.mdx:32]: There is a tip syntax ;-)

Comment (Rocketknight1) [reply docs/source/en/perf_train_tpu_tf.mdx:32]: Fixed!

**Review by sgugger — APPROVED** [review]
Thanks for the writeup, it looks very nice!

**Review by gante — APPROVED** [review]
🙏

**Review by sayakpaul — APPROVED** [review]
Since I have reviewed already, approving the content. Some minor todos:

- [x] Link the Colab Notebook from [here](https://github.com/huggingface/notebooks/pull/313). 

Also, are you not going to put the spaces you had added in the draft to leave some space for breathing? 🤣

**Comment by Rocketknight1** [comment]
@sayakpaul The spaces have been removed because the extra content after them got moved to a whole other doc!

**Pull request merged**